File: backend/app/schemas/finding.py
"""
Schemas Pydantic para Findings
"""
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel

from app.models.finding import FindingSeverity


class FindingResponse(BaseModel):
    """Schema para respuesta de finding"""
    id: UUID
    job_id: UUID
    severity: FindingSeverity
    title: str
    description: Optional[str] = None
    evidence: Optional[str] = None
    recommendation: Optional[str] = None
    tool: str
    created_at: datetime
    
    class Config:
        from_attributes = True

File: frontend/src/types/finding.ts
export type FindingSeverity = 'critical' | 'high' | 'medium' | 'low' | 'info'

export interface Finding {
  id: string
  job_id: string
  severity: FindingSeverity
  title: string
  description?: string | null
  evidence?: string | null
  recommendation?: string | null
  tool: string
  created_at: string
}
